pkg/controller/oam/applicationconfiguration/render.go

```diff
@@ -90,10 +95,17 @@ func (r *components) Render(ctx context.Context, ac *v1alpha2.ApplicationConfigu
 		if err != nil {
 			return nil, errors.Wrapf(err, errFmtRenderWorkload, acc.ComponentName)
 		}
-
+		w.SetNamespace(ac.GetNamespace())
 		ref := metav1.NewControllerRef(ac, v1alpha2.ApplicationConfigurationGroupVersionKind)
 		w.SetOwnerReferences([]metav1.OwnerReference{*ref})
-		w.SetNamespace(ac.GetNamespace())
+		// add child resources lists to the workload CR as annotation
+		crMap, err := r.fetchChildResourcesFromWorkloadDefinition(ctx, w)
```

Comment:
If the `ApplicationConfiguration` controller sees and renders everything, do the `Workload` need these annotations? Can the `ApplicationConfiguration` controller just refuse to render the workloads and traits if they are not compatible?

Reply:
We will remove this part per discussion. We need to bring back the OAM admission controller in v1alpha1.